codegen/src/main/java/software/amazon/awssdk/codegen/internal/Utils.java

```diff
@@ -77,10 +78,24 @@ public static boolean isOrContainsEnumShape(Shape shape, Map<String, Shape> allS
 
     public static boolean isOrContainsEnum(MemberModel member) {
         boolean isEnum = member.getEnumType() != null;
-        boolean isMapWithEnumMember = member.isMap() && (member.getMapModel().getKeyModel().getEnumType() != null ||
-                                                         member.getMapModel().getValueModel().getEnumType() != null);
-        boolean isListWithEnumMember = member.isList() && member.getListModel().getListMemberModel().getEnumType() != null;
-        return isEnum || isMapWithEnumMember || isListWithEnumMember;
+        return isEnum || isMapWithEnumShape(member) || isListWithEnumShape(member);
+    }
+
+    public static boolean isListWithEnumShape(MemberModel member) {
+        return member.isList() && member.getListModel().getListMemberModel().getEnumType() != null;
+    }
+
+    public static boolean isMapWithEnumShape(MemberModel member) {
+        return member.isMap() && (member.getMapModel().getKeyModel().getEnumType() != null ||
```

Comment:
Let's replace the 2nd and 3rd expressions with `isMapKeyWithEnumShape` and `isMapValueWithEnumShape`

Reply:
Done